cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -142,6 +143,13 @@ type NodeConfiguration struct {
 	// without CA verification via DiscoveryTokenCACertHashes. This can weaken
 	// the security of kubeadm since other nodes can impersonate the master.
 	DiscoveryTokenUnsafeSkipCAVerification bool
+
+	// FeatureGates enabled by the user
+	FeatureGates map[string]bool
+}
+
+type KubeletConfiguration struct {
+	BaseConfig *kubeletconfig.KubeletConfiguration
```

Comment:
I'd caution against using the internal representation of the KubeletConfiguration. Unlike the external versions, it's unversioned and carries no compatibility guarantees.

Reply:
Agreed. Done. Thanks for pointing out!